qt/MakeDialog.cc

```diff
@@ -258,14 +256,14 @@ MakeDialog::dragEnterEvent (QDragEnterEvent * event)
 {
   const QMimeData * mime = event->mimeData ();
 
-  if (!mime->urls ().isEmpty () && QFileInfo (mime->urls ().front ().path ()).exists ())
+  if (!mime->urls ().isEmpty () && QFileInfo (mime->urls ().at (0).path ()).exists ())
     event->acceptProposedAction ();
 }
 
 void
 MakeDialog::dropEvent (QDropEvent * event)
 {
-  const QString filename = event->mimeData ()->urls ().front ().path ();
+  const QString filename = event->mimeData ()->urls ().at (0).path ();
```

Comment:
Along with `for` -> `foreach` changes, this one looks so wrong and totally counter-intuitive... I bet it won't be long until the next person sending in the patch will call `first()` or `front()` instead of `at(0)`, or `last()` or `back()` instead of `at(size()-1)`. The fact that Qt containers [could] detach in all those cases for me is more of an implementation detail, and I don't like the idea of writing ugly[-ish] code just to workaround this. In this particular case, I'd declare a `const QList<QUrl>` (so it won't detach) variable and check for container size before calling any of the mentioned functions which could assert; this would've been much better change, but then again it has nothing to do with Qt 5 and C++11.